pkg/appgateserver/config/appgate_configs_reader_test.go

```diff
@@ -0,0 +1,115 @@
+package config_test
+
+import (
+	"net/url"
+	"testing"
+
+	sdkerrors "cosmossdk.io/errors"
+	"github.com/gogo/status"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/appgateserver/config"
+	"github.com/pokt-network/poktroll/testutil/yaml"
+)
+
+func Test_ParseAppGateConfigs(t *testing.T) {
+	tests := []struct {
+		desc     string
+		err      *sdkerrors.Error
+		expected *config.AppGateConfig
+		config   string
```

Comment:
What do you think of reordering and refactoring it like so? I had to do this translation in my head so I don't think others would complain.
```suggestion
		desc     string

		inputConfig   string

		expectedErr      *sdkerrors.Error
		expectedConfig *config.AppGateConfig
```